### PR TITLE
Feature/edit image details

### PR DESCRIPTION
**Pull request by aerych**
Adds feature for editing image details. See https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/427 for implementation discussion. 

Sample:
ios simulator screen shot jan 20 2015 8 49 02 pm

### CODE REVIEW

**Comment by aerych** [comment]
cc @bummytime

**Comment by bummytime** [comment]
Nice job @aerych. As always, I give you some initial general comments based purely on usage. :grinning:

The first big one is being able to select an image while the editor is in preview mode. Ideally, no kind of image selection should be allowed. We talked about this offline and decided to make another issue that will allow us to prevent this behavior. I only mention it here for posterity. Let me know if I should create said issue.

Secondly is that (IMHO) the transition for the actual edit overlay seems abrupt. This is just my opinion and feel free to ignore. :wink: 

Other than those two things, I like the editing capability and it worked like a champ during my initial testing.

Review of code coming next...

**Comment by aerych** [comment]
> Secondly is that (IMHO) the transition for the actual edit overlay seems abrupt.

I wondered about that, and was hoping for this specific feedback. I chose the cross dissolve intentionally since we've used it in different places in the past, but I wasn't sure if the default modal transition would be better.  I'll switch it to the standard.

File: WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+#import "UIImageView+AFNetworkingExtra.h"
+#import "WPTableViewController.h"
+#import "WPTableViewSectionHeaderView.h"
+
+static NSString *const TextFieldCell = @"TextFieldCell";
+static NSString *const CellIdentifier = @"CellIdentifier";
+static NSString *const ThumbCellIdentifier = @"ThumbCellIdentifier";
+static CGFloat CellHeight = 44.0f;
+
+typedef enum {
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m]: I'll only mention it once... NS_ENUM instead? Not sure if we landed on something official for the style guide. Besides you code style is so good that I have to find something to nitpick :ok_hand:

Comment (aerych) [inline WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m]: Arg! I do enums so infrequently I forget our convention! Will update.

**Comment by bummytime** [comment]
I poured through this code and it really seems pretty tight. Looks like you already have a fix for the preview taps here: https://github.com/wordpress-mobile/WordPress-iOS-Editor/pull/492

I'll check that out and maybe we can integrate that into this PR?

**Comment by aerych** [comment]
> I'll check that out and maybe we can integrate that into this PR?

I've already updated the podfile with the proper commit ;) Just waiting to pull the trigger.

**Comment by bummytime** [comment]
Success! :shipit:

**Comment by aerych** [comment]
:boom: Thanks for all the reviews and testing @bummytime !!!

**Pull request merged**